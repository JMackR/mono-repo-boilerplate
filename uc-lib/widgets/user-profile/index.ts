export * from "./user-profile"
export * from "./user-profile-types"
export * from "./user-profile-shared"
export * from "./public-user-profile"
export * from "./user-profile.constants"
