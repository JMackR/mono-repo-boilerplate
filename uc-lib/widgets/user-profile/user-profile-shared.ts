import moment from "moment"
export const memberSince = (joinDate: string) => {
  const joined = moment(joinDate, "YYYYMMDDmm").format("MMM YYYY")
  return `Member since ${joined}`
}
