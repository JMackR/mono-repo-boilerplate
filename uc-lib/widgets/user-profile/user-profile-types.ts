export interface ProfileRatingPropTypes {
  count: number
  average: number
}

export interface ProfilePropTypes {
  id: string
  avatar: string
  name: string
  rating: ProfileRatingPropTypes
  joined: string
  isAutosDealer?: boolean
  isMerchant?: boolean
  isSubPrimeDealer?: boolean
  tagLine?: string
}

export interface UserProfilePropTypes {
  profile: ProfilePropTypes
  avatarSize?: number
  onClick?(): void
  /**
   * Used to locate this view in end-to-end tests.
   */
  testID?: string
}
