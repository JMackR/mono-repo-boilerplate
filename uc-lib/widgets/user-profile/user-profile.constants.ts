export const AVATAR_SIZE = 48
export const MARGIN_USER_CONTENT = 4
export const MARGIN_STAR_RATING_CONTAINER = 1
export const AVATAR_SIZE_LARGE = 96
export const AVATAR_SIZE_MEDIUM = 74
export const DEALER_BADGE_OFFSET_STEP = 11
export const AVATAR_SIZE_EXTRA_LARGE = 160
