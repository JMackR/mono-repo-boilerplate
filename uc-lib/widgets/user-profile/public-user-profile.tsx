import React from "react"
import { Margin, Avatar, Flex, Stack, Text, IconBadge } from "uc-lib"

import { PublicProfilePropTypes } from "./public-user-profile.props"
import { AVATAR_SIZE } from "./user-profile.constants"
import { DealerBadge } from "../../assets"
import _ from "lodash"

export const PublicUserProfile: React.FC<PublicProfilePropTypes> = (props) => {
  const { avatars, isTruyouVerified, isAutosDealer, name, publicLocationName, ratingSummary, testID } = props

  const { average = 0, count = 0 } = ratingSummary || {}
  const avatarImageUri: string = _.property<any, string>("squareImage")(avatars)
  return (
    <Flex grow={1} testID={testID || "uc-lib.public-user-profile"}>
      <Flex>
        <Margin marginStep={2}>
          <Avatar source={{ uri: avatarImageUri || "" }} size={AVATAR_SIZE} testID="redibs-ucl.public-user-profile.avatar" />
        </Margin>
      </Flex>
      <Flex grow={1}>
        <Margin marginStep={2}>
          <Stack direction="column" childSeparationStep={1} crossAxisDistribution="flex-start">
            <Text testID="redibs-ucl.public-user-profile.first-name">{name}</Text>
            <Text textType="tertiaryBody2" color="secondary" testID="redibs-ucl.public-user-profile.public-location-name">
              {publicLocationName}
            </Text>
            {count > 0 && (
              <Flex direction="row">
                <Margin marginLeftStep={1}>
                  <Text textType="tertiaryBody2" color="secondary" testID="redibs-ucl.public-user-profile.rating-count">
                    ({count})
                  </Text>
                </Margin>
              </Flex>
            )}
          </Stack>
        </Margin>
      </Flex>
    </Flex>
  )
}
