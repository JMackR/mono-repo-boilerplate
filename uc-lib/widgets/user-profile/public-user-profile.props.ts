import { ProfileRatingPropTypes } from "./user-profile-types"

export interface PublicProfilePropTypes {
  avatars?: {
    squareImage: string
  }
  name: string
  ratingSummary: ProfileRatingPropTypes
  publicLocationName: string
  isTruyouVerified: boolean
  isAutosDealer: boolean
  /**
   * Used to locate this view in end-to-end tests.
   */
  testID?: string
  showLocation?: boolean | undefined
  showRating?: boolean | undefined
}
