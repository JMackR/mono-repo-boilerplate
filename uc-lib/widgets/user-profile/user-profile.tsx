import React from "react"
import { UserProfilePropTypes } from "./user-profile-types"
import { ActionRightChevron } from "uc-lib/assets"
import { Avatar, Text, SVG, Margin, Stack } from "uc-lib/controls"

import { StyleSheet, css } from "aphrodite"
import { AVATAR_SIZE, MARGIN_USER_CONTENT, MARGIN_STAR_RATING_CONTAINER } from "./user-profile.constants"
import { memberSince } from "./user-profile-shared"

const STYLES = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
    ":hover": {
      cursor: "pointer",
    },
  },
  userContent: {
    flexGrow: 1,
    whiteSpace: "nowrap",
  },
})

export const UserProfile: React.FC<UserProfilePropTypes> = ({ profile, onClick }) => {
  const memberSinceText = profile.tagLine || memberSince(profile.joined)
  return (
    <div className={css(STYLES.container)} onClick={onClick}>
      <Avatar source={{ uri: profile.avatar }} size={AVATAR_SIZE} />
      <div className={css(STYLES.userContent)}>
        <Margin marginLeftStep={MARGIN_USER_CONTENT}>
          <Stack direction="column">
            <Text>{profile.name}</Text>
            <Text textType="tertiaryBody2" color="secondary">
              {memberSinceText}
            </Text>
          </Stack>
        </Margin>
      </div>
      <SVG tint="hint" localSVG={ActionRightChevron} />
    </div>
  )
}
