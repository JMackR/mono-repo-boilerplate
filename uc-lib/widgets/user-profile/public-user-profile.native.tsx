import { Avatar, Flex, Margin, Stack, Text, Center } from "uc-lib"
import React from "react"
import { PublicProfilePropTypes } from "./public-user-profile.props"
import { AVATAR_SIZE } from "./user-profile.constants"

const SMALL_MARGIN = 0.5

export const PublicUserProfile: React.FC<PublicProfilePropTypes> = (props) => {
  const { avatars, name, publicLocationName, ratingSummary, testID, showLocation, showRating } = props
  const { average = 0, count = 0 } = ratingSummary || {}
  const showLocationSection = showLocation || showLocation === undefined ? true : false
  const showRatingSection = (showRating || showRating === undefined) && count > 0 ? true : false
  return (
    <Flex grow={1} testID={testID || "uc-lib.public-user-profile"}>
      <Flex>
        <Margin marginStep={2}>
          <Avatar
            source={{ uri: avatars?.squareImage || "" }}
            size={AVATAR_SIZE}
            testID="redibs-ucl.public-user-profile.avatar"
          />
        </Margin>
      </Flex>
      <Flex grow={1}>
        <Margin marginStep={2}>
          <Center>
            <Stack direction="column" childSeparationStep={1} crossAxisDistribution="flex-start">
              <Text testID="redibs-ucl.public-user-profile.first-name">{name}</Text>
              {showLocationSection && (
                <Text
                  textType="tertiaryBody2"
                  color="secondary"
                  testID="redibs-ucl.public-user-profile.public-location-name"
                >
                  {publicLocationName}
                </Text>
              )}
              {showRatingSection && (
                <Flex direction="row">
                  <Margin marginTopStep={SMALL_MARGIN}>
                    <StarRating rating={average} disabled={true} testID="redibs-ucl.public-user-profile.star-rating" />
                    <Margin marginLeftStep={1}>
                      <Text textType="tertiaryBody2" color="secondary" testID="redibs-ucl.public-user-profile.rating-count">
                        ({count})
                      </Text>
                    </Margin>
                  </Margin>
                </Flex>
              )}
            </Stack>
          </Center>
        </Margin>
      </Flex>
    </Flex>
  )
}
