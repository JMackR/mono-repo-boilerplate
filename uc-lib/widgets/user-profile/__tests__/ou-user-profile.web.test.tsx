import React from "react"
import { shallow, ShallowWrapper } from "enzyme"
import renderer from "react-test-renderer"
import { UserProfile } from "../user-profile"
import { UserProfilePropTypes } from "../user-profile-types"

describe("<UserProfile />", () => {
  const createComponent = (props: UserProfilePropTypes) => <UserProfile {...props} />
  const createWrapper = (props: UserProfilePropTypes) => shallow(<UserProfile {...props} />)

  let wrapper: ShallowWrapper, props: UserProfilePropTypes
  beforeEach(() => {
    const profile = {
      name: "Terrence",
      avatar: "https://static.redibs-stg.com/images/default_bg.png",
      joined: "2015-08-01T00:00:00+0000",
      rating: {
        count: 22,
        average: 3.4,
      },
    }
    props = {
      profile,
      onClick: jest.fn(),
    }
    wrapper = createWrapper(props)
  })
  it("should contain relavant component", () => {
    expect(wrapper.html()).toEqual(expect.stringContaining("Member since Aug 2015"))
  })
  it("should trigger event if clicking", () => {
    wrapper.simulate("click")
    expect(props.onClick).toHaveBeenCalled()
  })
  describe("Snapshot", () => {
    it("should render properly", () => {
      const tree = renderer.create(createComponent(props))
      expect(tree).toMatchSnapshot()
    })
  })
})
