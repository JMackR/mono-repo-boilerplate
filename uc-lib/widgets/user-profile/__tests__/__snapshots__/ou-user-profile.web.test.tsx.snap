// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> Snapshot should render properly 1`] = `
.container {
  display: flex !important;
  align-items: center !important;
}

.container:hover {
  cursor: pointer !important;
}

.userContent {
  flex-grow: 1 !important;
  white-space: nowrap !important;
}

.ouflex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  flex-wrap: undefined;
  align-items: stretch;
  justify-content: flex-start;
  max-width: 100vw;
  width: 48px;
  height: 48px;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.container {
  position: relative;
  justify-content: center;
  align-items: center;
  display: block;
  width: 48px;
  height: 0px;
  border-radius: 24px;
  padding-top: 100%;
  overflow: hidden;
  min-width: 48px;
}

.image {
  position: absolute;
  max-width: 100%;
  top: 0px;
  bottom: 0px;
  height: 100%;
  width: 100%;
  object-fit: cover;
  display: block;
}

.children {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 2;
}

.oumargin {
  margin: undefined;
  margin-left: 16px;
  margin-right: undefined;
  margin-top: undefined;
  margin-bottom: undefined;
  display: flex;
  position: relative;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  max-width: undefined;
  width: undefined;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: column;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.text-textStyle {
  color: #121212;
  font-size: 16px;
  font-weight: bold;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

.oustack {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  width: undefined;
  max-width: undefined;
  overflow: undefined;
}

.oumargin {
  margin: undefined;
  margin-left: undefined;
  margin-right: 4px;
  margin-top: undefined;
  margin-bottom: undefined;
  display: flex;
  position: relative;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  max-width: undefined;
  width: undefined;
}

.starImageStyle {
  width: 12px;
  height: 12px;
  cursor: default;
}

.root {
  height: undefined;
  width: 4px;
  display: inline-block;
}

.text-textStyle {
  color: #8A8A8A;
  font-size: 12px;
  font-weight: normal;
  font-family: Lato, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
}

<div
  className="container"
  onClick={[MockFunction]}
>
  <div
    className="ouflex"
  >
    <picture
      className="container"
    >
      <source
        srcSet="https://static.redibs-stg.com/images/default_bg.png"
      />
      <img
        className="image"
        onLoad={[Function]}
        src="https://static.redibs-stg.com/images/default_bg.png"
      />
      <div
        className="children"
      />
    </picture>
  </div>
  <div
    className="userContent"
  >
    <div
      className="oumargin"
    >
      <div
        className="oustack"
      >
        <div
          className="text-textStyle"
        >
          Terrence
        </div>
        <div
          className="oustack"
        >
          <div
            className="oumargin"
          >
            <div
              className="oustack"
            >
              <div
                className="starImageStyle"
                onClick={[Function]}
              >
                <rating-fill-full-star.svg
                  alwaysdark="#121212"
                  alwayslight="#FFFFFF"
                  basalt="#8A8A8A"
                  crystal="#FFFFFF"
                  crystalhover="#F7F7F7"
                  crystalpressed="#EEEEEE"
                  disabled="#FAFAFA"
                  glacialblue="#3D98D2"
                  granite="#ABABAB"
                  height={12}
                  larchyellow="#F4BE19"
                  larchyellowhover="#F5D624"
                  larchyellowpressed="#F3A70C"
                  limestone="#E2E2E2"
                  limestonehover="#EEEEEE"
                  limestonepressed="#D0D0D0"
                  obsidian="#121212"
                  overlay="rgba(18,18,18,0.3)"
                  paintbrushred="#E05666"
                  palousegreen="#00A87E"
                  palousegreenhighlight="#DEF3EC"
                  palousegreenhover="#43B893"
                  palousegreenpressed="#00986A"
                  quartz="#FAFAFA"
                  width={12}
                />
              </div>
              <div
                className="root"
              />
              <div
                className="starImageStyle"
                onClick={[Function]}
              >
                <rating-fill-full-star.svg
                  alwaysdark="#121212"
                  alwayslight="#FFFFFF"
                  basalt="#8A8A8A"
                  crystal="#FFFFFF"
                  crystalhover="#F7F7F7"
                  crystalpressed="#EEEEEE"
                  disabled="#FAFAFA"
                  glacialblue="#3D98D2"
                  granite="#ABABAB"
                  height={12}
                  larchyellow="#F4BE19"
                  larchyellowhover="#F5D624"
                  larchyellowpressed="#F3A70C"
                  limestone="#E2E2E2"
                  limestonehover="#EEEEEE"
                  limestonepressed="#D0D0D0"
                  obsidian="#121212"
                  overlay="rgba(18,18,18,0.3)"
                  paintbrushred="#E05666"
                  palousegreen="#00A87E"
                  palousegreenhighlight="#DEF3EC"
                  palousegreenhover="#43B893"
                  palousegreenpressed="#00986A"
                  quartz="#FAFAFA"
                  width={12}
                />
              </div>
              <div
                className="root"
              />
              <div
                className="starImageStyle"
                onClick={[Function]}
              >
                <rating-fill-full-star.svg
                  alwaysdark="#121212"
                  alwayslight="#FFFFFF"
                  basalt="#8A8A8A"
                  crystal="#FFFFFF"
                  crystalhover="#F7F7F7"
                  crystalpressed="#EEEEEE"
                  disabled="#FAFAFA"
                  glacialblue="#3D98D2"
                  granite="#ABABAB"
                  height={12}
                  larchyellow="#F4BE19"
                  larchyellowhover="#F5D624"
                  larchyellowpressed="#F3A70C"
                  limestone="#E2E2E2"
                  limestonehover="#EEEEEE"
                  limestonepressed="#D0D0D0"
                  obsidian="#121212"
                  overlay="rgba(18,18,18,0.3)"
                  paintbrushred="#E05666"
                  palousegreen="#00A87E"
                  palousegreenhighlight="#DEF3EC"
                  palousegreenhover="#43B893"
                  palousegreenpressed="#00986A"
                  quartz="#FAFAFA"
                  width={12}
                />
              </div>
              <div
                className="root"
              />
              <div
                className="starImageStyle"
                onClick={[Function]}
              >
                <rating-fill-half-star.svg
                  alwaysdark="#121212"
                  alwayslight="#FFFFFF"
                  basalt="#8A8A8A"
                  crystal="#FFFFFF"
                  crystalhover="#F7F7F7"
                  crystalpressed="#EEEEEE"
                  disabled="#FAFAFA"
                  glacialblue="#3D98D2"
                  granite="#ABABAB"
                  height={12}
                  larchyellow="#F4BE19"
                  larchyellowhover="#F5D624"
                  larchyellowpressed="#F3A70C"
                  limestone="#E2E2E2"
                  limestonehover="#EEEEEE"
                  limestonepressed="#D0D0D0"
                  obsidian="#121212"
                  overlay="rgba(18,18,18,0.3)"
                  paintbrushred="#E05666"
                  palousegreen="#00A87E"
                  palousegreenhighlight="#DEF3EC"
                  palousegreenhover="#43B893"
                  palousegreenpressed="#00986A"
                  quartz="#FAFAFA"
                  width={12}
                />
              </div>
              <div
                className="root"
              />
              <div
                className="starImageStyle"
                onClick={[Function]}
              >
                <rating-fill-no-star.svg
                  alwaysdark="#121212"
                  alwayslight="#FFFFFF"
                  basalt="#8A8A8A"
                  crystal="#FFFFFF"
                  crystalhover="#F7F7F7"
                  crystalpressed="#EEEEEE"
                  disabled="#FAFAFA"
                  glacialblue="#3D98D2"
                  granite="#ABABAB"
                  height={12}
                  larchyellow="#F4BE19"
                  larchyellowhover="#F5D624"
                  larchyellowpressed="#F3A70C"
                  limestone="#E2E2E2"
                  limestonehover="#EEEEEE"
                  limestonepressed="#D0D0D0"
                  obsidian="#121212"
                  overlay="rgba(18,18,18,0.3)"
                  paintbrushred="#E05666"
                  palousegreen="#00A87E"
                  palousegreenhighlight="#DEF3EC"
                  palousegreenhover="#43B893"
                  palousegreenpressed="#00986A"
                  quartz="#FAFAFA"
                  width={12}
                />
              </div>
            </div>
          </div>
          <div
            className="text-textStyle"
          >
            (
            22
            )
          </div>
        </div>
        <div
          className="text-textStyle"
        >
          Member since Aug 2015
        </div>
      </div>
    </div>
  </div>
  <actions-chevron-right-line.svg
    alwaysdark="#121212"
    alwayslight="#FFFFFF"
    basalt="#8A8A8A"
    crystal="#FFFFFF"
    crystalhover="#F7F7F7"
    crystalpressed="#EEEEEE"
    disabled="#FAFAFA"
    glacialblue="#3D98D2"
    granite="#ABABAB"
    larchyellow="#F4BE19"
    larchyellowhover="#F5D624"
    larchyellowpressed="#F3A70C"
    limestone="#E2E2E2"
    limestonehover="#EEEEEE"
    limestonepressed="#D0D0D0"
    obsidian="#121212"
    overlay="rgba(18,18,18,0.3)"
    paintbrushred="#E05666"
    palousegreen="#00A87E"
    palousegreenhighlight="#DEF3EC"
    palousegreenhover="#43B893"
    palousegreenpressed="#00986A"
    quartz="#FAFAFA"
    tintcolor="#ABABAB"
  />
</div>
`;
